src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/DetectorCollection.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
+{
+    public static class ResponseDetectorViewExtensions
+    {
+        public static DiagnosticData AddDetectorCollection(this Response response, List<string> detectorIds)
```

Comment:
How does this work in the UI. Do you show a loading message for every detector? 
If yes, Where does the loading message comes from? It cant be from detector response. So it has to be defined here. 
Or are you using the name of the detector?

Reply:
I had a more detailed conversation with Hawk last week, I can share with you what my vision is as well.